strawberry/experimental/pydantic/utils.py

```diff
@@ -24,3 +26,15 @@ def get_strawberry_type_from_model(type_: Any):
         return type_._strawberry_type
     else:
         raise UnregisteredTypeException(type_)
+
+
+def get_private_fields(cls: Type) -> List[dataclasses.Field]:
+    private_fields: List[dataclasses.Field] = []
+    for field in dataclasses.fields(cls):
+        if isinstance(field.type, Private):
+            private_fields.append(field)
+    return private_fields
+
+
+class auto:
```

Comment:
I wasn't sure if this was the best way to do this, but it does seem to work and gives us some options for adding things to the object if needed.

Reply:
I think we should start using the new sentinel class: https://www.python.org/dev/peps/pep-0661/

It's not currently in the standard lib but they have a reference implementation that should be easy to include: https://github.com/taleinat/python-stdlib-sentinels